mne/io/eeglab/eeglab.py

```diff
@@ -226,6 +233,12 @@ class RawEEGLAB(_BaseRaw):
         Names or indices of channels that should be designated
         EOG channels. If 'auto', the channel names containing
         ``EOG`` or ``EYE`` are used. Defaults to empty tuple.
+    event_id : dict
+        By default, events are read from the input file by dropping every
+        non-integer part of events containing integers, and completely
+        dropping any events without integer parts. If non-integer events
+        should be read, this should be a dict mapping from their names to
+        integers, e.g. dict(fmri_scan_onset=199, recording_start=255).
```

Comment:
I had missed the callable idea. I like it too. I would call it event_id_func

the logging should explicit how each event str is mapped to what int

we should make it consistent between all readers.

the default behavior sounds fine to me.